### DISCUSSION TITLE
Same segment multiple times, different settings

### DISCUSSION THREAD

**Discussion by Syphdias** [romkatv/powerlevel10k#2784] [General]
Is there a way to have a segment twice and have different settings for it? for example a `vcs_clone1` that can uses `POWERLEVEL9K_VCS_CLONE2_*` instead of `POWERLEVEL9K_VCS_*`.

I know you can treat it differently with `p10k display` depending on the position. I don't think making a simple wrapper function like this would work:
```zsh
function prompt_vcs_clone2() {
    prompt_vcs "$@"
}
```

I feel like I have seen this before…
This might be overkill for https://github.com/romkatv/powerlevel10k/issues/2781 but I was still curious if there is a simple generic way to achieve this.

**Comment by romkatv** [comment]
There is no supported way to do this.

What are you trying to achieve?

**Reply by Syphdias** [reply]
It was just a curiosity of mine, I have no use for it. I think the feeling that I had seen something like this was just from `p10k display` being able to treat the same segment at another position differently (e.g. time on the left for the active prompt but after pressing enter it would "switch" to the right).

My first idea for #2781 was to use another vcs segement and use a different content expansion and  `POWERLEVEL9K_VCS_CLONE2_SHOW_ON_COMMAND` for only one of the segments. That way it might be more performant than just making a custom segment that calls `p10k segment -t $(git rev-parse --abbrev-ref HEAD)`.

**Reply by romkatv** [reply]
The `p10k display` API is fairly decent because it was designed from scratch. Much of the challenge of working on p10k was keeping its configuration backward-compatible with p9k. As you recall, p9k grew organically and the first attempt of a backward incompatible clean slate redesign has been abandoned before it saw the light of the day. p10k kept going without a backward-incompatible change for a very long time, and it has taken its toll on the usability of configs. Making them even more complex is not something I would like to see.

> My first idea for #2781 was to use another vcs segement

Your suggestion on that issue is a good one. A custom segment can use `VCS_STATUS_LOCAL_BRANCH` without doing anything special with gitstatus. These parameters will be provided by the primary vcs prompt segment. This makes the implementation of the custom segment fairly straightforward, and it'll be blazingly fast.

**Reply by Syphdias** [reply]
I had wild ideas of coping `prompt_vcs` and helper functions, but I had no intention of keeping the pieces, if I break it ;)

> A custom segment can use `VCS_STATUS_LOCAL_BRANCH` without doing anything special with gitstatus.

I was unsure if it was accessible in a custom segment, so I didn't suggest it.

**Reply by romkatv** [reply]
> I was unsure if it was accessible in a custom segment, so I didn't suggest it.

It's not documented but I suggested this approach in issues before, so I know people rely on it. I won't break it.
